.github/workflows/ci.yml

```
@@ -0,0 +1,43 @@
+name: Node CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
+        os: [ubuntu-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      # Checkout the npm/cli repo
+      - uses: actions/checkout@v2
+
+        # Installs the specific version of Node.js
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+        # Run the installer script
+      - name: Install dependencies
+        run: npm ci
+
+        # Run the tests
+      - name: Run tests
+        run: npm t
+
+      # Run coverage check
+      - name: Run Coverage Report
+        if: matrix.node-version == '12.x'
+        run: npm run coverage
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_OPTIONAL_TOKEN }}
```

nit: I would change this from the magic `tap` environment var to the actual one that `coveralls` cli uses.
```suggestion
          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
```